File: src/main/java/committee/nova/quit/util/Utilities.java
package committee.nova.quit.util;

import committee.nova.quit.client.key.KeyInit;
import org.lwjgl.input.Keyboard;

public class Utilities {
    private static boolean isClosing = false;

    private static boolean isAnyKeyDown(int... keys) {
        for (final int key : keys) if (Keyboard.isKeyDown(key)) return true;
        return false;
    }

    public static boolean direct() {
        return isAnyKeyDown(KeyInit.fastQuit.getKeyCode(), KeyInit.bossKey.getKeyCode());
    }

    public static boolean boss() {
        return isAnyKeyDown(KeyInit.bossKey.getKeyCode());
    }

    public static boolean fast() {
        return isAnyKeyDown(KeyInit.fastQuit.getKeyCode());
    }

    public static void markClosing() {
        isClosing = true;
    }

    public static boolean isClosing() {
        return isClosing;
    }
}

File: src/main/scala/committee/nova/quit/util/Utilities.scala
package committee.nova.quit.util

import committee.nova.quit.client.key.KeyInit
import org.lwjgl.input.Keyboard

object Utilities {
  var isClosing = false

  private def isAnyKeyDown(keys: Int*): Boolean = {
    keys.foreach(id => {
      if (Keyboard.isKeyDown(id)) return true
    })
    false
  }

  def direct: Boolean = isAnyKeyDown(KeyInit.fastQuit.getKeyCode, KeyInit.bossKey.getKeyCode)

  def boss: Boolean = isAnyKeyDown(KeyInit.bossKey.getKeyCode)

  def fast: Boolean = isAnyKeyDown(KeyInit.fastQuit.getKeyCode)
}
